python/lsst/daf/butler/datastores/sasquatchDatastore.py

```diff
@@ -0,0 +1,885 @@
+            A string which will be used in creating unique identifier tags.
+        identifierFields: `Mapping` or `None`
+            The keys and values in this mapping will be both added as fields
+            in the record, and used in creating a unique tag for the uploaded
+            dataset type. I.e. the timestamp, and the tag will be unique, and
+            each record will belong to one combination of such.
+        extraFields: `Mapping`
+            Extra mapping keys and values that will be added as fields to the
+            dispatched record.
+
+        Returns
+        -------
+        result : `Mapping` of `str` to `list`
+            A mapping of metric name of list of metric measurement records
+        status : `bool`
+            A status bool indicating if some records had to be skipped due to
+            a problem parsing the bundle.
+        """
+        if id is None:
+            id = uuid4()
+        sid = str(id)
+        meta: dict[str, Any] = dict()
+
+        # Add other associated common fields
+        meta["id"] = sid
+        meta["run"] = run
+        meta["dataset_type"] = datasetType
+
+        # Check to see if the bundle declares a reference package
+        self._handleReferencePackage(meta, bundle)
+
+        # Handle the various timestamps that could be associated with a record
+        self._handleTimes(meta, bundle, run)
+
+        # Always use the supplied timestamp if one was passed to use.
+        if timestamp is not None:
+            meta["timestamp"] = timestamp.timestamp()
+
+        self._handleIdentifier(meta, identifierFields, datasetIdentifier, bundle)
+
+        # Add in any other fields that were supplied to the function call.
+        if extraFields is not None:
+            meta.update(extraFields)
+
+        metricRecords: dict[str, list[Any]] = dict()
+
+        # Record if any records needed skipped
+        resultsTrimmed = False
+
+        # Look at each of the metrics in the bundle (name, values)
+        for metric, measurements in bundle.items():
+            # create a list which will contain the records for each measurement
+            # associated with metric
+            metricRecordList = metricRecords.setdefault(metric, list())
+
+            record: dict[str, Any] = dict()
+            record.update(meta)
+
+            # loop over each metric measurement within the metric
+            for measurement in measurements:
+                # need to extract any tags, package info, etc
+                notes = dict(measurement.notes.items())
+                record["tags"] = notes.get("metric_tags", list())
+
+                # Missing values are replaced by 0 in sasquatch, see RFC-763
+                name = ""
+                value = 0.0
+                match measurement.json:
+                    case {"metric": name, "value": None}:
+                        pass
+                    case {"metric": name, "value": value}:
+                        if value != value:
```

Comment:
Should we be using the more explicit:
```python
if math.isnan(value):
```
rather than being this abstract?

Reply:
There was some edge case where something was not considering something nan, mixing type implementations or something like that, I cant remember what now though. We can switch to `math` and see if the edge case comes up again, I dont have strong feelings on it, outside this is exactly the definition of nan by the IEEE, and so I think it is a reasonable choice

Reply:
I do feel that `math.isnan` is more explicit and it is implemented in C so it shouldn't be slower than what you have. If you still want to do the `x != x` code please add a comment -- not every python programmer will understand what you mean there.